commands/bnet.js

```diff
@@ -0,0 +1,443 @@
+   */
+  get numBattleTags() {
+    return this.battleTags.size;
+  }
+
+  /**
+   * Compares two BattleTagEntry's and for sorting.
+   * @param {BattleTagEntry} a
+   * @param {BattleTagEntry} b
+   * @return {number} 0 if a and b are sorted equally, -1 if a should be sorted
+   * before b, and 1 if a should be sorted after b.
+   */
+  static compare(a, b) {
+    const cmpA = a.user ? a.user.displayName : a.textId;
+    const cmpB = b.user ? b.user.displayName : b.textId;
+    if (cmpA === cmpB) return 0;
+    return cmpA < cmpB ? -1 : 1;
+  }
+
+  /**
+   * @return {string} Returns the Snowflake string of the stored user if one
+   * exists, or the text ID otherwise.
+   */
+  toIdString() {
+    if (this.user) {
+      return this.user.toString();
+    }
+    return this.textId;
+  }
+
+  /**
+   * @return {string} Returns a pretty-printed line of text containing the ID
+   * string and a list of BattleTags.
+   */
+  toEntryString() {
+    return this.toIdString() + ': ' +
+           Array.from(this.battleTags).sort().join(', ');
+  }
+}
+
+/**
+ * Retrieves the specified channel using the specified client.
+ * @param {Discord.Client} client The client to use for querying
+ * @param {string} channelId The snowflake for the channel to retrieve
+ * @return {Discord.Channel} The channel retrieve, or null if it does not
+ * exist.
+ */
+function getChannelById(client, channelId) {
+  if (!client) {
+    console.error('Error getting channel: No cilent specified');
+    return null;
+  }
+
+  if (!channelId) {
+    console.error('Error getting channel: No channel ID specified');
+    return null;
+  }
+
+  const channel = client.channels.get(channelId);
+  if (!channel) {
+    console.error(`Error getting channel: Could not find channel with ` +
+                  `ID ${channelId}`);
+  }
+  return channel;
+};
+
+/**
+ * Adds all BattleTags in the BattleTags list to the entry for the specified
+ * user or text key. User keys are their Snowflake ID. If an entry was storing
+ * a text ID and a user is specified the text ID is overriden and the list of
+ * BattleTags is reset.
+ * @param {string} key The key of the user to add too
+ * @param {(Discord.GuildMember|string)} userOrTextId The GuildMember or text ID
+ * to store
+ * @param {string[]} battleTagList A list of BattleTags to add
+ */
+function addBattleTags(key, userOrTextId, battleTagList) {
+  let discordUser = null;
+  let textId = null;
+  if (userOrTextId instanceof Discord.GuildMember) {
+    discordUser = userOrTextId;
+  } else {
+    textId = userOrTextId;
+  }
+  const debugString = discordUser ? discordUser.toString() : 'null';
+  console.log(`Add BattleTag: ${key}, ${debugString}, ${textId}: ` +
+              `${battleTagList}`);
+  if (!battleTags.has(key)) {
+    battleTags.set(key, new BattleTagEntry(discordUser, textId, battleTagList));
+  } else {
+    const existingEntry = battleTags.get(key);
+    if (!discordUser && existingEntry.user) {
```

Comment:
Do we just silently fail if a user tries to add a text id but the tagentry uses a guildmember?

Reply:
No longer an issue as this has been removed